Backend/src/controllers/auth.controller.ts

```diff
@@ -0,0 +1,102 @@
+import { Request, Response } from 'express';
+import bcrypt from 'bcryptjs';
+import jwt from 'jsonwebtoken';
+import { User, UserInput } from '../models/user.model';
+import { Token } from '../models/token.model';
+
+const generateToken = (userId: string, secret: string, expiresIn: string) => {
+  return jwt.sign({ id: userId }, secret, { expiresIn });
+};
+
+const register = async (req: Request, res: Response) => {
+  const { email, password, fullName, role } = req.body;
+  if (!email || !fullName || !password) {
+    return res.status(422).json({ message: 'The fields email, fullNamea and password are required' });
+  }
+
+  try {
+    const user = await User.findOne({ email });
+    if (user) {
+      return res.status(400).send('User already exist');
+    }
+
+    const hashedPassword = await bcrypt.hash(password, 10);
+    const newUser: UserInput = {
+      email,
+      password: hashedPassword,
+      enabled: true,
+      fullName,
+      role: role,
+    };
+    const userCreated = await User.create(newUser);
+    res.status(201).json({ name: userCreated.fullName, email: userCreated.email, _id: userCreated.id });
```

Comment:
Change status code 201 to a constant

Reply:
HTTP status codes are defined in a separate constants file and referenced from there.